src/transformStateWithClones.js

```diff
@@ -8,6 +8,53 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let prevState = { ...state };
+
+  for (const action of actions) {
+    let stepResult = { ...prevState };
+
+    switch (action.type) {
+      case 'addProperties':
+        stepResult = {
+          ...stepResult,
+          ...action.extraData,
+        };
+
+        result.push(stepResult);
+
+        prevState = { ...stepResult };
+
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (stepResult[key]) {
+            delete stepResult[key];
+          }
+        }
+
+        result.push(stepResult);
+
+        prevState = { ...stepResult };
+
+        break;
+
+      case 'clear':
+        stepResult = {};
+
+        result.push(stepResult);
+
+        prevState = { ...stepResult };
```

Comment:
It is not necessary to create a variable under an empty object.
```suggestion
        result.push({});
```